File: Gateway/src/main/java/com/quizGpt/Gateway/Quiz/Entity/Quiz.java
package com.quizGpt.formManagement.Quiz.Entity;

import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@ToString
@Table(name = "Quizes")
public class Quiz {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(targetEntity = Question.class, cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name="quizId_fk", referencedColumnName = "id")
    private List<Question> Questions;
}

File: resource-server/src/quiz/entities/quiz.ts
import { Column, Entity, OneToMany, PrimaryColumn, Relation } from "typeorm";
import { Question } from "./question.js";
import { QuizAttempt } from "./quizAttempt.js";

@Entity("Quizes")
export class Quiz {
  @PrimaryColumn()
  quizId!: string;

  @Column()
  title!: string;

  @OneToMany(() => Question, (question) => question.quiz, {
    cascade: true,
    eager: true,
  })
  questions!: Relation<Question[]>;

  @OneToMany(() => QuizAttempt, (quizAttempt) => quizAttempt.quiz)
  attempts!: Relation<QuizAttempt[]>;

  //   @Column()
  //   @ManyToOne(() => User, (user) => user.quizzes)
  //   owner: User;
}
